packages/components/src/Tree/Tree.hooks.ts

```diff
@@ -120,3 +121,32 @@ function makeDetailedItems(
 		return base
 	})
 }
+
+
+// Construct a default expansion map by rolling up an expand for any parents of selected or expanded children to the top of the tree.
+// This is only used to initialize the expansion, so that user interactions are not overridden.
+function forceExpansionToSelected(items: TreeItem[], selectedKey?: string): Record<string, boolean> {
+	const collect = (children: TreeItem[], parentKey?: string): string[] => {
+		return children.map(child => {
+			if (child.children) {
+				const childKeys = collect(child.children, child.key)
+				if (childKeys && childKeys.length > 0) {
```

Comment:
does `childKeys?.length > 0` work?

Reply:
oh, it does now - I had previously returned undefined if no keys